### PR TITLE
🐛(tests) configure DRF in lms settings & use lms settings in tests

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

We need to use lms settings in our tests so that we are in an Open edX context.

## Proposal

- [x] Configure DRF URL path versioning in lms settings
- [x] Test allowed API version
- [x] Switch tests settings to lms docker settings
- [x] Populate test database in Fonzie test suite
- [x] Remove `tox`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -8,9 +8,11 @@ ENV DOCKERIZE_VERSION v0.6.0
...
 ARG group=0
 
-# Add a non-privileged user to run the application
-RUN groupadd --gid $group app && \
-    useradd --uid $user --gid $group --home /app --create-home app
+# Add a non-privileged user to run the application if given as a build argument
+RUN if [ ${user} -ne 0 -a ${group} -ne 0 ]; then \
+        groupadd --gid $group app ; \
+        useradd --uid $user --gid $group --home /app app ; \
+    fi
```
Comment (sampaccoud) [inline Dockerfile:15]: use redhat pattern as we don't want to create images that work only with one user?

File: .travis.yml [hunk 2]
```diff
@@ -5,10 +5,10 @@ cache:
...
   - node_modules
 env:
-- TOXENV=quality
-- TOXENV=docs
-- TOXENV=django18
-- API_SPEC=1 OPENEDX_RELEASE=ginkgo.1
+- CI_STEP=quality
+- CI_STEP=docs
+- CI_STEP=test
+- CI_STEP=test-spec OPENEDX_RELEASE=ginkgo.1
```
Comment (sampaccoud) [inline .travis.yml:11]: I thought you were removing Travis?

File: bin/ci [hunk 3]
```diff
@@ -59,49 +58,55 @@ function install_dependencies() {
...
+        echo "Loading database schema from cache: $sql"
+        # FIXME: docker-compose is having issues with huge files streamed
+        # via a pty/stdin, so we bypass compose to use raw docker exec
+        # command.
+        docker exec -i $(_docker_compose ps -q mysql) mysql -u root $database < $sql
     else
-        echo "Running tox:$TOXENV test suite..."
+        # Perform database migrations
+        echo "Starting database migration"
+        _dc_run lms python ./manage.py lms migrate
```
Comment (sampaccoud) [inline bin/ci:87]: cms migrations?

File: bin/ci [hunk 4]
```diff
@@ -59,49 +58,55 @@ function install_dependencies() {
...
+    case "$CI_STEP" in
+        quality)
+            make quality
+            ;;
+        docs)
+            make docs
+            ;;
+        test)
+            make test
+            ;;
```
Comment (cchaudier) [inline bin/ci]: Line 95 -> 103 ca be replaced by
docs|quality|test)
            make $CI_STEP
            ;;

File: Dockerfile (hunk 1 above)
Comment (jmaupetit) [reply Dockerfile:15]: If you don't mind I would prefer to handle this in a new PR as it requires substantial changes. I'll declare a new issue for this. Deal?

File: bin/ci (hunk 3 above)
Comment (jmaupetit) [reply bin/ci:87]: Fonzie is installed in the LMS application. Is the LMS standalone, or does it requires the CMS to run?

File: bin/ci (hunk 4 above)
Comment (jmaupetit) [reply bin/ci]: Nice catch! Thx!

File: .travis.yml (hunk 2 above)
Comment (jmaupetit) [reply .travis.yml:11]: I will. Not in this PR. I'll declare another issue for this. Deal?

Comment (jmaupetit) [reply .travis.yml:11]: FTR https://github.com/openfun/fonzie/issues/10

File: Dockerfile (hunk 1 above)
Comment (jmaupetit) [reply Dockerfile:15]: FTR #11

File: bin/ci (hunk 3 above)
Comment (sampaccoud) [reply bin/ci:87]: ok so maybe remove it from elsewhere?

**Review by sampaccoud — APPROVED** [review]

**Review by cchaudier — APPROVED** [review]

**Pull request merged**